tools/clang/include/clang/SPIRV/Type.h

```diff
@@ -0,0 +1,118 @@
+//===-- Type.h - SPIR-V Type --*- C++-*------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_CLANG_SPIRV_TYPE_H
+#define LLVM_CLANG_SPIRV_TYPE_H
+
+#include <set>
+#include <unordered_set>
+#include <vector>
+
+#include "clang/SPIRV/Decoration.h"
+#include "clang/SPIRV/Option.h"
+#include "clang/SPIRV/spirv.hpp"
+
+namespace clang {
+namespace spirv {
+
+class SPIRVContext;
+
+class Type {
+public:
+  spv::Op getOpcode() const { return opcode; }
+  const std::vector<uint32_t> &getArgs() const { return args; }
+  bool isBooleanType(const Type *t);
```

Comment:
yeah... i don't know how this happened! looks like an old version of the code got mixed in :)